activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -40,7 +40,7 @@ def arel_visitor # :nodoc:
       self.emulate_booleans = true
 
       NATIVE_DATABASE_TYPES = {
-        primary_key: "int auto_increment PRIMARY KEY",
+        primary_key: "BIGINT(8) UNSIGNED DEFAULT NULL auto_increment PRIMARY KEY",
```

Comment:
This should not have `DEFAULT NULL`. It is invalid in MySQL 5.7 https://github.com/rails/rails/pull/13247